src/start_fact.py

```diff
@@ -71,60 +72,37 @@ def _terminate_process(process: Popen):
             os.kill(process.pid, signal.SIGUSR1)
             process.wait(timeout=60)
         except TimeoutExpired:
-            logging.error('component did not stop in time -> kill')
             process.kill()
         except ProcessLookupError:
             pass
 
 
 def shutdown(*_):
-    logging.info('shutting down...')
-    fact.run = False
-
-
-def _process_is_running(process: Popen) -> bool:
-    try:
-        os.kill(process.pid, 0)
-        if process.poll() is not None:
-            return False
-        return True
-    except ProcessLookupError:
-        return False
+    global run
```

Comment:
This may very well be what the global statement was originally intended for. I would still argue (for the reasons mentioned above) that if we *can* avoid using it, we should do so. We should probably convert it to an instance variable and move the `shutdown()` function into the class, though.